File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums, target):
        nm={}
        for i, n in enumerate(nums):
            a = target-n
            if a in nm:
                return [nm[a], i]
            nm[n]=i
if __name__ == "__main__":
    sol = Solution()
    print(sol.twoSum([2, 7, 11, 15], 9))  
    print(sol.twoSum([3, 2, 4], 6))       
    print(sol.twoSum([3, 3], 6))          

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& n, int t){
        for(int i=0;i<n.size();i++){
            for(int j=i+1;j<n.size();j++){
                if(n[i]+n[j]==t){
                    return{i,j};
                }
            }
        }
        return { 0,0};
    }
};